spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -234,6 +234,13 @@ private static void apply(UpdateSchema pendingUpdate, TableChange.AddColumn add)
         add.isNullable(),
         "Incompatible change: cannot add required column: %s",
         leafName(add.fieldNames()));
+    if (add.defaultValue() != null) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Cannot add column %s since default values are currently unsupported",
+              leafName(add.fieldNames())));
+    }
```

Comment:
[doubt][not a blocker]  A while back we had a thread to align with Spark's standardized error handling (doc : https://docs.google.com/document/d/11qHUiCcKMJ-xAyfL__Yv7B1b5N-80-GIwE8AV96A2Ac/edit?tab=t.0) if we are ok 
can we throw this documented error class then https://github.com/apache/spark/blob/master/sql/api/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala#L731

Reply:
Thanks for the doc, I'll double check if there's some standard error code but I don't really agree with the linked error class since that's specific to query parsing failures (e.g. [some tests](https://github.com/apache/spark/pull/35690/files#diff-b9e91f767e5562861565b0ce78759af3bcb7fff405a81e928894641147db2ae4R2243) in the original Spark PR demonstrate the intention) whereas the intention of the implemented error is to surface that the Iceberg-Spark implementation just doesn't support this yet.